mirrord/agent/src/file.rs

```diff
@@ -159,6 +160,14 @@ impl FileManager {
                 let access_result = self.access(pathname.into(), mode);
                 Ok(FileResponse::Access(access_result))
             }
+            FileRequest::Lstat(LstatRequest { path }) => {
+                let pathname = path
```

Comment:
Anyway to avoid this duplication with Access? Nest the match arms?

Reply:
it will create another level of nesting..
it's also copy-paste of the stuff above so tbh we need more macros.

Reply:
Do we think nesting is somehow worse than having duplicate code? And what's the stuff above?

Reply:
Having match inside match to share 2 lines of code seems like bad choice, especially as two structs have nothing in common aside from that 2-line-long-prolog.
This whole place needs to be revamped with generics +macros so I don't think that's worth doing now